packages/server/graphql/mutations/helpers/pushEstimateToGitHub.ts

```diff
@@ -0,0 +1,226 @@
+      name: repoName,
+      owner: repoOwner,
+      number: issueNumber
+    })
+    if (issueResError) return issueResError
+    const {repository} = issueRes
+    if (!repository) return new Error('Repository not found on GitHub')
+
+    const {issue} = repository
+    if (!issue) return new Error('Issue not found on GitHub')
+
+    const {id: issueId} = issue
+    const {name: meetingName, phases} = meeting
+    const estimatePhase = getPhase(phases, 'ESTIMATE')
+    const {stages} = estimatePhase
+    const stageIdx = stages.findIndex((stage) => stage.id === stageId)
+    const discussionURL = makeAppURL(appOrigin, `meet/${meetingId}/estimate/${stageIdx + 1}`)
+    const body = makeScoreGitHubComment(
+      dimensionName,
+      value || '<None>',
+      meetingName,
+      discussionURL
+    )
+
+    const [, commentError] = await githubRequest<AddCommentMutation, AddCommentMutationVariables>(
+      addComment,
+      {
+        input: {
+          body,
+          subjectId: issueId
+        }
+      }
+    )
+    if (commentError) return commentError
+    return undefined
+  }
+
+  const [repoLabelsRes, repoLabelsError] = await githubRequest<
+    GetRepoLabelsQuery,
+    GetRepoLabelsQueryVariables
+  >(getRepoLabels, {
+    name: repoName,
+    owner: repoOwner,
+    issueNumber,
+    first: 100
+  })
+  if (repoLabelsError) return repoLabelsError
+
+  const {repository} = repoLabelsRes
+  if (!repository) return new Error('Repository not found on GitHub')
+
+  const {id: repositoryId, labels, issue} = repository
+  if (!labels) return new Error('Labels for repository not found')
+  if (!issue) return new Error(`Issue #${issueNumber} not found`)
+
+  const {id: issueId, labels: issueLabels} = issue
+  const {nodes} = labels
+  if (!nodes) return new Error('Label nodes not found')
+  if (!issueLabels) return new Error('Issue labels not found')
+
+  const {nodes: nullableIssueLabelNodes} = issueLabels
+  if (!nullableIssueLabelNodes) return new Error('Issue label nodes not found')
+
+  const issueLabelNodes = nullableIssueLabelNodes.filter(isNotNull)
+  const labelNodes = nodes.filter(isNotNull)
+  const matchingLabel = labelNodes.find((node) => node.name === githubLabelName)
+  const DESCRIPTION_IDENTIFIER = 'Generated by Parabol'
+  let labelToAddId = matchingLabel?.id ?? ''
+  if (!matchingLabel) {
+    let color = PALETTE.GRAPE_500.slice(1)
+    if (meeting) {
+      const {templateRefId} = meeting
+      const templateRef = await dataLoader.get('templateRefs').load(templateRefId)
+      const {dimensions} = templateRef
+      const dimensionRef = dimensions.find((dimension) => dimension.name === dimensionName)
+      if (!dimensionRef) return new Error('Dimension not found')
+      const {scaleRefId} = dimensionRef
+      const scaleRef = await dataLoader.get('templateScaleRefs').load(scaleRefId)
+      const {values} = scaleRef
+      const matchingValue = values.find((scaleValue) => scaleValue.label === value)
```

Comment:
A small bug I ran into:
- create a label called `Points` and click Update
- edit the label by adding an empty space at the end
- update the Story Points final score and click Update
- see the error message: `Name has already been taken`

It looks like GitHub trims the labels so we'll need to do the same. Currently, if the Parabol & GitHub labels are identical except for whitespace, `matchingValue` is undefined so we try to create a new label, GitHub trims it, and then complains that it already exists

Reply:
top notch find :fire: